### PR TITLE
RavenDB-22279 - ClusterWideTransaction_WhenRestoreFromIncrementalBackupAfterStoreAndDelete_ShouldDeleteInTheDestination fails on sharding

### PR DESCRIPTION
**Pull request by shaharhikri**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-22279/SlowTests.Issues.RavenDB21050.ClusterWideTransactionWhenRestoreFromIncrementalBackupAfterStoreAndDeleteShouldDeleteInTheDestinat

### Additional description

SaveChanges for ClusterWideTransaction on sharded db is waiting for index notification only on the ClusterStateMachine and not on the shards, so the backup is ending before the shareds perform the cluster-wide transactions commands.
On restore, we don't need to wait because the cluster transactions are executed directly on the shards `DocumentDatabase` for each backup file.

### Type of change

- [x] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [ ] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [x] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [ ] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

File: src/Raven.Server/Documents/Sharding/Handlers/Processors/Batches/ShardedClusterTransactionRequestProcessor.cs [hunk 1]
```diff
@@ -27,12 +28,13 @@ public ShardedClusterTransactionRequestProcessor(ShardedDatabaseRequestHandler r
 
     public override Task WaitForDatabaseCompletion(Task<HashSet<string>> onDatabaseCompletionTask, long index, ClusterTransactionOptions options, CancellationToken token)
     {
-        return Task.CompletedTask;
+        var op = new ShardedWaitForIndexNotificationOperation(RequestHandler, index);
+        return RequestHandler.ShardExecutor.ExecuteParallelForAllAsync(op, token);
```
Comment (karmeli87) [inline src/Raven.Server/Documents/Sharding/Handlers/Processors/Batches/ShardedClusterTransactionRequestProcessor.cs]: use `WaitForExecutionOnShardsAsync(index)` instead

Comment (karmeli87) [reply src/Raven.Server/Documents/Sharding/Handlers/Processors/Batches/ShardedClusterTransactionRequestProcessor.cs]: note, that we need to add a token to that method

**Review by karmeli87 — CHANGES_REQUESTED** [review]
Please check failing tests

File: src/Raven.Server/Documents/Sharding/Handlers/Processors/Batches/ShardedClusterTransactionRequestProcessor.cs (hunk 1 above)
Comment (shaharhikri) [reply src/Raven.Server/Documents/Sharding/Handlers/Processors/Batches/ShardedClusterTransactionRequestProcessor.cs]: Done

File: test/SlowTests/Issues/RavenDB-21050.cs [hunk 2]
```diff
@@ -62,6 +69,60 @@ public async Task ClusterWideTransaction_WhenRestoreFromIncrementalBackupAfterSt
                 }
             }
 
+        }
+    }
+
+
+    [RavenTheory(RavenTestCategory.BackupExportImport)]
```
Comment (karmeli87) [inline test/SlowTests/Issues/RavenDB-21050.cs]: Sharding as well

**Review by karmeli87 — APPROVED** [review]

File: test/SlowTests/Issues/RavenDB-21050.cs (hunk 2 above)
Comment (shaharhikri) [reply test/SlowTests/Issues/RavenDB-21050.cs]: Done

**Pull request merged**